diagnosticsModule/Public/Join-DiagnosticsFiles.ps1

```diff
@@ -0,0 +1,106 @@
+<#
+.SYNOPSIS
+Merges one or more diagnostics output files into a single merged file named MergedDiagnosticsFile.json.  This merged file is used with the Diagnostics Analyzer Tool on the AD FS Help website (https://adfshelp.microsoft.com/DiagnosticsAnalyzer).
+
+.DESCRIPTION
+The Join-DiagnosticsFiles cmdlet is used to merge one or more diagnostics output files into a single file.  You can then upload this merged file to AD FS Help and view the farm health and configuration holistically.
+
+.PARAMETER FilePath
+The location of the output files that will be merged
+
+.EXAMPLE
+Join-DiagnosticsFiles c:\output
+Merge the diagnostics files located in the c:\output folder
+
+.EXAMPLE
+Join-DiagnosticsFiles .
+Merge the diagnostics files located in the current folder
+#>
+Function Join-DiagnosticsFiles
+{
+    [CmdletBinding()]
+    Param
+    (
+        [string] $FilePath = $null
+    )
+
+    # Merged data file name
+    $FileFilter = "ADFSDiagnosticsFile*.json"
+    $OutputFileName = "\MergedDiagnosticsFile.json"
+
+    # Dictionary of the objects used to create the merged test collection
+    $mergedTestData = @{}
+    $mergedTestData["AllTests"] = @()
+    $mergedTestData["PassedTests"] = @()
+    $mergedTestData["WarningTests"] = @()
+    $mergedTestData["FailedTests"] = @()
+    $mergedTestData["ErrorTests"] = @()
+    $mergedTestData["NotRunTests"] = @()
+    $mergedTestData["ReachableServers"] = @()
+    $mergedTestData["UnreachableServers"] = @()
+
+    # Objects to hold the version and ADFS Configuration information
+    $version =""
+    $adfsConfiguration = ""
+
+    # Read each file in the folder
+    Write-Host "Locating diagnostics files at: $FilePath" 
+
+    $files = Get-ChildItem $FilePath -Filter $FileFilter
+
+    # Make sure that at least one file exists
+    if ($files.Length -eq 0) {
+        Write-Error "There are no diagnostics files found at: '$FilePath'.  Make sure the path is correct and the files are in format '$FileFilter'."
+        return
+    }
+
+    $files | 
+        ForEach-Object {
+            
+            $fileName = $_ | Select-Object -ExpandProperty FullName
+
+            Write-Host "  - Merging file: $fileName"  
+            
+            $allTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select AllTests
```

Comment:
make part of this into a shared variable so that we aren't calling Get-Content, ConvertFrom-Json, SelectObject multiple times